SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/WorkingSetSystem.cpp

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "LoadBalancing/WorkingSetSystem.h"
+
+#include "Algo/AllOf.h"
+#include "Algo/Copy.h"
+
+#include "Interop/WorkingSetsCommon.h"
+#include "Schema/WorkingSet.h"
+#include "SpatialView/ComponentData.h"
+#include "SpatialView/EntityComponentTypes.h"
+#include "SpatialView/EntityDelta.h"
+#include "SpatialView/SubView.h"
+#include "SpatialView/ViewCoordinator.h"
+
+DEFINE_LOG_CATEGORY_STATIC(LogSpatialWorkingSetSystem, Log, All);
+
+namespace SpatialGDK
+{
+void FWorkingSetSystem::Advance(const FSubView& Subview)
+{
+	NewMembers.Empty();
+	EntitiesToRefresh.Empty();
+	NewMarkers.Empty();
+	FreedEntities.Empty();
+
+	for (const EntityDelta& Delta : Subview.GetViewDelta().EntityDeltas)
+	{
+		switch (Delta.Type)
+		{
+		case EntityDelta::ADD:
+		{
+			// This is a marker entity that requires updating.
+			const EntityViewElement& Entity = Subview.GetView()[Delta.EntityId];
+
+			FWorkingSetCommonData Data(Entity);
+
+			check(Data.Request.Generation >= Data.Response.Generation);
+
+			const bool bCanApproveNewActorSet = Algo::AllOf(
+				Data.Request.RequestedEntities, [this, MarkerEntityId = Delta.EntityId](const Worker_EntityId_Key RequestedEntityId) {
+					const Worker_EntityId* RequestedEntityOwner = ActorOwnership.Find(RequestedEntityId);
+					return RequestedEntityOwner == nullptr || *RequestedEntityOwner == MarkerEntityId;
+				});
+
+			check(bCanApproveNewActorSet);
+			if (bCanApproveNewActorSet)
+			{
+				NewMarkers.Emplace(Delta.EntityId);
+				EntitiesToRefresh.Emplace(Data.Request.LeaderEntityId);
```

Comment:
Only the leader needs to be refreshed as all member entities will join it irregardless of their LB data.